core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -243,7 +246,17 @@ public void configure() {
          if (rvrl != null) {
             rvrl.remoteValueNotFound(key);
          }
-         return null;
+         if (!hasSuccesfulResponse) {
+            RpcException rpcException = new RpcException("No successful response: " + responses);
+            for (Response r : responses.values()) {
+               if (r instanceof ExceptionResponse) {
+                  rpcException.addSuppressed(((ExceptionResponse) r).getException());
+               }
+            }
+            throw rpcException;
```

Comment:
Right, upon exception, the exception is thrown directly from JGroupsTransport so this loop is unnecessary.
On the other hand, Successful/UnsuccessfulResponse should be handled at interceptor-level, so the check for hasSuccessfulReponse makes sense.